pkg/blocks/step.go

```diff
@@ -22,241 +22,176 @@ package blocks
 import (
 	"errors"
 	"fmt"
-	"runtime"
-	"strings"
 
 	"github.com/facebookincubator/ttpforge/pkg/logging"
-	"github.com/facebookincubator/ttpforge/pkg/outputs"
-	"go.uber.org/zap"
 	"gopkg.in/yaml.v3"
 )
 
-// Constants representing supported executor types.
-const (
-	ExecutorPython     = "python3"
-	ExecutorBash       = "bash"
-	ExecutorSh         = "sh"
-	ExecutorPowershell = "powershell"
-	ExecutorRuby       = "ruby"
-	ExecutorBinary     = "binary"
-	ExecutorCmd        = "cmd.exe"
-)
-
-// StepType denotes the type of a step in a TTP.
-type StepType string
-
-// Constants for defining the types of steps available.
-const (
-	StepCreateFile = "createFileStep"
-	StepUnset      = "unsetStep"
-	StepFile       = "fileStep"
-	StepFetchURI   = "fetchURIStep"
-	StepBasic      = "basicStep"
-	StepSubTTP     = "subTTPStep"
-	StepCleanup    = "cleanupStep"
-	StepEdit       = "editStep"
-)
+type CommonStepFields struct {
+	Name        string `yaml:"name,omitempty"`
+	Description string `yaml:"description,omitempty"`
 
-// Act represents a single action within a TTP (Tactics, Techniques,
-// and Procedures) step.
-//
-// Condition: The condition that needs to be satisfied for the Act to execute.
-// Environment: Environment variables used during the Act's execution.
-// Name: The unique name of the Act.
-// WorkDir: The working directory of the Act.
-// Type: The type of the Act (e.g., Command, File, or Setup).
-// success: Indicates whether the execution of the Act was successful.
-// stepRef: Reference to other steps in the sequence.
-// output: The output of the Act's execution.
-type Act struct {
```

Comment:
Act has been wholly removed for multiple reasons: 

* Common fields like `Name` are now centralized in `Step`, where they can share validation code rather than having their validation code duplicated in every step
* Most of the fields here (like `Environment`) are not actually common to all step types. Only `inline:`, `file:`, and `ttp:` care about Environment - `create_file` and similar steps do not. Those fields have therefore been moved to their respective action types